File: b2c-parent/b2c-core/src/main/java/org/javamaster/b2c/core/exception/BizException.java
package org.javamaster.b2c.core.exception;

import org.javamaster.b2c.core.enums.BizExceptionEnum;

/**
 * @author yudong
 * @date 2019/6/10
 */
public class BizException extends RuntimeException {
    private final BizExceptionEnum bizExceptionEnum;

    public BizException(BizExceptionEnum bizExceptionEnum) {
        super(bizExceptionEnum.getErrorMsg());
        this.bizExceptionEnum = bizExceptionEnum;
    }

    @Override
    public String toString() {
        return bizExceptionEnum.getErrorCode() + " " + bizExceptionEnum.getErrorMsg();
    }

    public BizExceptionEnum getBizExceptionEnum() {
        return bizExceptionEnum;
    }
}

File: fragmentlearning/app/src/main/java/org/javamaster/fragmentlearning/exception/BizException.kt
package org.javamaster.fragmentlearning.exception

import org.javamaster.fragmentlearning.data.model.ResultVo
import java.io.IOException

/**
 * @author yudong
 * @date 2019/9/22
 */
class BizException(val resultVo: ResultVo<out Any>) : IOException(resultVo.errorMsg) {
    var errorCode: Int? = resultVo.errorCode
}